test_gif fails with pillow 11.0.0

Hi!

After upgrading pillow to 11.0.0 we are seeing the following test fail with imageio 2.36.0 on Python 3.12:


```pytb
___________________________________ test_gif ___________________________________

self = <PIL.GifImagePlugin.GifImageFile image mode= size=32x42 at 0x7FFEB303C170>
fp = <_io.BufferedReader name='/build/pytest-of-nixbld/pytest-0/test_gif0/test0.0.0.gif'>
filename = ''

    def __init__(
        self, fp: StrOrBytesPath | IO[bytes], filename: str | bytes | None = None
    ) -> None:
        super().__init__()
    
        self._min_frame = 0
    
        self.custom_mimetype: str | None = None
    
        self.tile: list[_Tile] = []
        """ A list of tile descriptors, or ``None`` """
    
        self.readonly = 1  # until we know better
    
        self.decoderconfig: tuple[Any, ...] = ()
        self.decodermaxblock = MAXBLOCK
    
        if is_path(fp):
            # filename
            self.fp = open(fp, "rb")
            self.filename = os.path.realpath(os.fspath(fp))
            self._exclusive_fp = True
        else:
            # stream
            self.fp = cast(IO[bytes], fp)
            self.filename = filename if filename is not None else ""
            # can be overridden
            self._exclusive_fp = False
    
        try:
            try:
>               self._open()

/nix/store/lz34mqqjjdps6hzb6yfwyr744zwrfp0j-python3.12-pillow-11.0.0/lib/python3.12/site-packages/PIL/ImageFile.py:144: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/lz34mqqjjdps6hzb6yfwyr744zwrfp0j-python3.12-pillow-11.0.0/lib/python3.12/site-packages/PIL/GifImagePlugin.py:122: in _open
    self._seek(0)  # get ready to read first frame
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <PIL.GifImagePlugin.GifImageFile image mode= size=32x42 at 0x7FFEB303C170>
frame = 0, update_image = True

    def _seek(self, frame: int, update_image: bool = True) -> None:
        if frame == 0:
            # rewind
            self.__offset = 0
            self.dispose: _imaging.ImagingCore | None = None
            self.__frame = -1
            self._fp.seek(self.__rewind)
            self.disposal_method = 0
            if "comment" in self.info:
                del self.info["comment"]
        else:
            # ensure that the previous frame was loaded
            if self.tile and update_image:
                self.load()
    
        if frame != self.__frame + 1:
            msg = f"cannot seek to frame {frame}"
            raise ValueError(msg)
    
        self.fp = self._fp
        if self.__offset:
            # backup to last frame
            self.fp.seek(self.__offset)
            while self.data():
                pass
            self.__offset = 0
    
        s = self.fp.read(1)
        if not s or s == b";":
            msg = "no more images in GIF file"
>           raise EOFError(msg)
E           EOFError: no more images in GIF file

/nix/store/lz34mqqjjdps6hzb6yfwyr744zwrfp0j-python3.12-pillow-11.0.0/lib/python3.12/site-packages/PIL/GifImagePlugin.py:200: EOFError

The above exception was the direct cause of the following exception:

tmp_path = PosixPath('/build/pytest-of-nixbld/pytest-0/test_gif0')

    @deprecated_test
    def test_gif(tmp_path):
        fnamebase = str(tmp_path / "test")
    
        # The not-animated gif
    
        for isfloat in (False, True):
            for crop in (0, 1, 2):
                for colors in (0, 3, 4):
                    if colors > 1 and sys.platform.startswith("darwin"):
                        continue  # quantize fails, see also png
                    fname = fnamebase + "%i.%i.%i.gif" % (isfloat, crop, colors)
                    rim = get_ref_im(colors, crop, isfloat)
    
                    try:
                        imageio.imsave(fname, rim, format="GIF-PIL")
                    except ImportError:
                        pytest.xfail("New pillow version is no longer supported.")
    
>                   im = imageio.imread(fname, format="GIF-PIL")

tests/test_pillow_legacy.py:286: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
imageio/__init__.py:97: in imread
    return imread_v2(uri, format=format, **kwargs)
imageio/v2.py:360: in imread
    result = file.read(index=0, **kwargs)
imageio/core/legacy_plugin_wrapper.py:147: in read
    reader = self.legacy_get_reader(**kwargs)
imageio/core/legacy_plugin_wrapper.py:116: in legacy_get_reader
    return self._format.get_reader(self._request)
imageio/core/format.py:221: in get_reader
    return self.Reader(self, request)
imageio/core/format.py:312: in __init__
    self._open(**self.request.kwargs.copy())
imageio/plugins/pillow_legacy.py:285: in _open
    self._im = factory(self._fp, "")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <PIL.GifImagePlugin.GifImageFile image mode= size=32x42 at 0x7FFEB303C170>
fp = <_io.BufferedReader name='/build/pytest-of-nixbld/pytest-0/test_gif0/test0.0.0.gif'>
filename = ''

    def __init__(
        self, fp: StrOrBytesPath | IO[bytes], filename: str | bytes | None = None
    ) -> None:
        super().__init__()
    
        self._min_frame = 0
    
        self.custom_mimetype: str | None = None
    
        self.tile: list[_Tile] = []
        """ A list of tile descriptors, or ``None`` """
    
        self.readonly = 1  # until we know better
    
        self.decoderconfig: tuple[Any, ...] = ()
        self.decodermaxblock = MAXBLOCK
    
        if is_path(fp):
            # filename
            self.fp = open(fp, "rb")
            self.filename = os.path.realpath(os.fspath(fp))
            self._exclusive_fp = True
        else:
            # stream
            self.fp = cast(IO[bytes], fp)
            self.filename = filename if filename is not None else ""
            # can be overridden
            self._exclusive_fp = False
    
        try:
            try:
                self._open()
            except (
                IndexError,  # end of data
                TypeError,  # end of data (ord)
                KeyError,  # unsupported mode
                EOFError,  # got header but not the first frame
                struct.error,
            ) as v:
>               raise SyntaxError(v) from v
E               SyntaxError: no more images in GIF file

/nix/store/lz34mqqjjdps6hzb6yfwyr744zwrfp0j-python3.12-pillow-11.0.0/lib/python3.12/site-packages/PIL/ImageFile.py:152: SyntaxError
```

--- a/tests/test_pillow_legacy.py
+++ b/tests/test_pillow_legacy.py
@@ -283,7 +283,11 @@ def test_gif(tmp_path):
                 except ImportError:
                     pytest.xfail("New pillow version is no longer supported.")
 
-                im = imageio.imread(fname, format="GIF-PIL")
+                try:
+                    im = imageio.imread(fname, format="GIF-PIL")
+                except SyntaxError:
+                    pytest.xfail("New pillow version is no longer supported.")
+
                 mul = 255 if isfloat else 1
                 if colors not in (0, 1):
                     im = im[:, :, :3]
@@ -331,8 +335,10 @@ def test_animated_gif(test_images, tmp_path):
             except ImportError:
                 pytest.xfail("Pillow version no longer supported.")
             # Retrieve
-            print("fooo", fname, isfloat, colors)
-            ims2 = imageio.mimread(fname, format="GIF-PIL")
+            try:
+                ims2 = imageio.mimread(fname, format="GIF-PIL")
+            except SyntaxError:
+                pytest.xfail("New pillow version is no longer supported.")
             ims1 = [x[:, :, :3] for x in ims]  # fresh ref
             ims2 = [x[:, :, :3] for x in ims2]  # discart alpha
             for im1, im2 in zip(ims1, ims2):

